lms/views/api/groups.py

```diff
@@ -1,15 +1,27 @@
+from marshmallow import Schema
 from pyramid.view import view_config
+from webargs import fields
 
 from lms.security import Permissions
+from lms.validation._base import PyramidRequestSchema
+
+
+class APIGroupSetsSchema(PyramidRequestSchema):
+    class LMS(Schema):
+        product = fields.Str(required=True)
+
+    context_id = fields.Str(required=True)
 
 
 @view_config(
     request_method="POST",
     permission=Permissions.API,
     renderer="json",
     route_name="api.courses.group_sets.list",
+    schema=APIGroupSetsSchema,
 )
 def course_group_sets(_context, request):
-    return request.product.plugin.grouping.get_group_sets(
-        request.matchdict["course_id"]
+    course = request.find_service(name="course").get_by_context_id(
+        context_id=request.parsed_params["context_id"]
```

Comment:
Nothing now on request.matchdict

Reply:
Why do we look-up the course here? Most plugins so far don't want the object, just the raw `context_id`. This seems like we've added a query in most places that we don't have to?

Reply:
We'll soon will use `course` in the plugin itself so I wanted to tackled both the change in the parameter to change the interface.

In retrospect his might not be as clear as the intent is not present in this PR.